components/ide/jetbrains/gateway-plugin/src/main/kotlin/io/gitpod/jetbrains/gateway/GitpodConnectionProvider.kt

```diff
@@ -144,6 +148,22 @@ class GitpodConnectionProvider : GatewayConnectionProvider {
             connectionPanel.repaint()
         }
 
+        val connectionKeyId = "${connectParams.gitpodHost}-${connectParams.workspaceId}-${connectParams.backendPort}"
+
+        if (activeConnections.containsKey(connectionKeyId)) {
+            val message = "You are trying to connect to a workspace that has a session already open"
+            val title = "Oops"
+            val okButton = Messages.getOkButton()
+            val options = arrayOf(okButton)
+            val defaultIndex = 0
+            val icon = Messages.getWarningIcon()
+            Messages.showDialog(message, title, options, defaultIndex, icon)
+
+            val errMessage = "A connection to the same workspace already exists: $connectionKeyId"
+            thisLogger().warn(errMessage)
+            throw IllegalArgumentException(errMessage)
```

Comment:
IllegalArgumentException - means that invalid argument was passed

maybe IllegalStateException is better here, not sure